### PR TITLE
Support vendored google protobuf sources.

### PR DESCRIPTION
**Pull request by corps**
Generated files are distributed with protobuf libraries, so they do not need to be built even if they are referenced.

Follow up PR creates dependency on `google.protobuf.struct`

### CODE REVIEW

File: buf.yaml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 version: v2
 modules:
   - path: proto
+    excludes:
+      - proto/google/protobuf
```
Comment (corps) [inline buf.yaml:5]: No need to actually build these files in practice, as the generated message types are distributed as part of our python runtime anyways.

File: buf.yaml [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 breaking:
   use:
     - FILE
   ignore_unstable_packages: true
-lint:
   ignore:
-    - "kafka/events.v1.proto"
-    - "relay/event.proto"
+    - google/proto
```
Comment (corps) [inline buf.yaml:11]: Also no need to lint these files.

File: Makefile [hunk 3]
```diff
@@ -30,5 +30,18 @@ build-rust:
 clean-rust:
 	cd rust && cargo clean
 
+repodir := $(dir $($(abspath $(lastword $(MAKEFILE_LIST)))))
+.PHONY: update-vendor
+update-vendor:
+	cd $$(mktemp -d) && \
```
Comment (corps) [inline Makefile:36]: Vendoring is just a shallow checkout and directory replacement.

File: Makefile [hunk 4]
```diff
@@ -30,5 +30,18 @@ build-rust:
...
+		git clone -n --depth=1 --filter=tree:0 \
+		    https://github.com/protocolbuffers/protobuf && \
+		cd protobuf && git sparse-checkout set --no-cone src/google/protobuf && \
+		git checkout && rm -rf .git && \
+		find . '(' ! -name '*.proto' -a ! -name '*.md' ')' -delete && \
+		find . && \
+		rm -rf $(repodir)proto/google && \
+		mv src/google $(repodir)proto/
```
[the remark this answers is not in the document]
Comment (corps) [reply Makefile:44]: Fair!  I'll keep pruning then.

**Review by markstory — APPROVED** [review]
Looks good to me.

**Pull request merged**